koku/masu/processor/tasks.py

```diff
@@ -494,3 +511,99 @@ def autovacuum_tune_schema(schema_name):
                     LOG.info(cursor.statusmessage)
 
     LOG.info(f"Altered autovacuum_vacuum_scale_factor on {alter_count} tables")
+
+
+@app.task(  # noqa: C901
+    name="masu.celery.tasks.convert_to_parquet",
+    queue_name="reporting",
+    autoretry_for=(ClientError,),
+    max_retries=10,
+    retry_backoff=10,
+)
+def convert_to_parquet(request_id, account, provider_uuid, provider_type, start_date, manifest_id, context={}):
+    """
+    Convert archived CSV data from our S3 bucket for a given provider to Parquet.
+
+    This function chiefly follows the download of a providers data.
+
+    This task is defined to attempt up to 10 retries using exponential backoff
+    starting with a 10-second delay. This is intended to allow graceful handling
+    of temporary AWS S3 connectivity issues because it is relatively important
+    for us to convert the archived data.
+
+    Args:
+        request_id (str): The associated request id (ingress or celery task id)
+        account (str): The account string
+        provider_uuid (UUID): The provider UUID
+        start_date (str): The report start time (YYYY-mm-dd)
+        manifest_id (str): The identifier for the report manifest
+        context (dict): A context object for logging
+
+    """
+    if not context:
+        context = {"account": account, "provider_uuid": provider_uuid}
+
+    if not request_id or not account or not provider_uuid:
+        if not request_id:
+            message = "missing required argument: request_id"
+            LOG.error(message)
+        if not account:
+            message = "missing required argument: account"
+            LOG.error(message)
+        if not provider_uuid:
+            message = "missing required argument: provider_uuid"
+            LOG.error(message)
+        if not provider_type:
+            message = "missing required argument: provider_type"
+            LOG.error(message)
+        return
+
+    if not settings.ENABLE_S3_ARCHIVING:
```

Comment:
Maybe move this to the top of this function so we don't log any error messages if `ENABLE_S3_ARCHIVING` is turned off?

Reply:
Done.